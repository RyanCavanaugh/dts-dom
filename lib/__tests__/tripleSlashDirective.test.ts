import {ContextFlags, TripleSlashDirective, create, emit} from "../index"

describe("an emitted module with triple slash directives", () => {
    it("matches the snapshot", () => {
        const tripleSlashDirectives: TripleSlashDirective[] = [];

        tripleSlashDirectives.push(create.tripleSlashReferencePathDirective("./test"));
        tripleSlashDirectives.push(create.tripleSlashReferenceTypesDirective("test"));
        tripleSlashDirectives.push(create.tripleSlashReferenceNoDefaultLibDirective());
        tripleSlashDirectives.push(create.tripleSlashAmdModuleDirective());
        tripleSlashDirectives.push(create.tripleSlashAmdModuleDirective("test"));

        const module = create.module("test");
        const emitOptions = {tripleSlashDirectives};

        expect(emit(module, emitOptions)).toMatchSnapshot();
    });
});

describe("an emitted module without triple slash directives", () => {
    it("matches the snapshot", () => {
        const module = create.module("test");

        expect(emit(module)).toMatchSnapshot();
    });
});
