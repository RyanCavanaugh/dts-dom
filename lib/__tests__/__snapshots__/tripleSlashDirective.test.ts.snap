// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an emitted module with triple slash directives matches the snapshot 1`] = `
/// <reference path="./test" />
/// <reference types="test" />
/// <reference no-default-lib="true" />
/// <amd-module />
/// <amd-module name="test" />
declare module 'test' {
}


`;

exports[`an emitted module without triple slash directives matches the snapshot 1`] = `
declare module 'test' {
}


`;
